src/hooks/use-heatmap/heatmap-fns.js

```diff
@@ -0,0 +1,38 @@
+export const weekdays = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday'];
+export const weekdayMap = weekdays.reduce((map, curr) => ({ ...map, [curr]: [] }), {});
+
+export const updateWeekdayHours = (hours, post) => {
+  const { hour, id } = post;
+  const foundHour = hours.find((h) => h.hour === hour);
+
+  if (foundHour) {
+    const { hour: hr, count, postIds } = foundHour;
+
+    return [
+      ...hours.filter((h) => h.hour !== hr),
+      {
+        hour: hr,
+        count: count + 1,
+        postIds: postIds.concat(id),
+      },
+    ];
+  }
+
+  return [...hours, { hour, count: 1, postIds: [id] }];
+};
+
+export const postsByWeekdayAndHour = (posts) => posts.map((p) => {
+  const date = new Date(p.data.created_utc * 1000);
+
+  return {
+    id: p.data.id,
+    weekday: weekdays[date.getDay()],
+    hour: date.getHours(),
+  };
+});
+
+export const getHeatmapFromPosts = (posts) => postsByWeekdayAndHour(posts)
```

Comment:
I'd consider working with a 2D or 3D array here. It'll make things a lot easier. After all, weekday and hour are already numbers that you can easily use as array index. Have a look at the example implementation. You'll find more information there.

Reply:
Ah, that's a much simpler approach. I will look to implement something like:

```js
[
  // Sunday
  [
    // 12am
    [
      // posts at 12am
    ]
  ] 
]
```

(I will reference your example implementation after I give this a try myself)